### ISSUE TITLE
[java] Renderers output wrong class qualified name for nested classes

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#4904]
**Affects PMD Version:** 7.0.0

**Rule:** [AbstractClassWithoutAbstractMethod](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#abstractclasswithoutabstractmethod)

**Description:** The AbstractClassWithoutAbstractMethod reports the wrong class

**Code Sample demonstrating the issue:**

```java
package com.github.cowwoc.lumina;

public final class Form
{
	public static abstract class Property    // <----- Rule is triggering on this line
	{
		// ...
	}

	// ...
}
```
(https://github.com/cowwoc/lumina/blob/main/java/src/main/java/com/github/cowwoc/lumina/Form.java#L38)

**Expected outcome:** Violation message with class name `com.github.cowwoc.lumina.Form.Property`
**Actual outcome:** `[INFO] PMD Failure: com.github.cowwoc.lumina.Property:38 Rule:AbstractClassWithoutAbstractMethod Priority:3 This abstract class does not have any abstract methods.`

**Running PMD through:** Maven

**Comment by cowwoc** [comment]
PS: The rule is triggering on this line: `private final JsonNode value;`

**Comment by oowekyala** [comment]
The rule is actually triggered on the following place: https://github.com/cowwoc/lumina/blob/main/java/src/main/java/com/github/cowwoc/lumina/Form.java#L38

There is a problem with the renderer as it just takes the simple name of the class and prepends the package name. This of course is wrong when the type is a nested class like here. The name of the class is `com.github.cowwoc.lumina.Form.Property` and not `com.github.cowwoc.lumina.Property`. Since you have the latter in the package the renderer output is confusing.

I think the renderer should output file names and not class names, for instance:
```
src/main/java/com/github/cowwoc/lumina/Form.java:38 Rule:AbstractClassWithoutAbstractMethod Priority:3 This abstract class does not have any abstract methods.
```
instead of the current output now (broken)
```
com.github.cowwoc.lumina.Property:38 Rule:AbstractClassWithoutAbstractMethod Priority:3 This abstract class does not have any abstract methods.
```
or "fixed", but it is not obvious which file the line number is for:
```
com.github.cowwoc.lumina.Form.Property:38 Rule:AbstractClassWithoutAbstractMethod Priority:3 This abstract class does not have any abstract methods.
```

**Comment by oowekyala** [comment]
Here is the problem in the Maven renderer:

https://github.com/apache/maven-pmd-plugin/blob/c54170707cfd0aec4b5d9ee6083230f2551cdaa4/src/main/java/org/apache/maven/plugins/pmd/PmdViolationCheckMojo.java#L96-L104

But other renderers in PMD have the same problem, eg 

https://github.com/pmd/pmd/blob/e5f55c08a5b2fb31820bca04fb23416cdd884611/pmd-core/src/main/java/net/sourceforge/pmd/renderers/TextColorRenderer.java#L217-L219

**Comment by jsotuyod** [comment]
@adangel I'm unsure this holds for the 3.22.0 plugin version, but you may want to take this to the maven plugin's JIRA.

**Comment by adangel** [comment]
For nested classes, what are the expected values in additionalinfo of rule violations?

https://github.com/pmd/pmd/blob/6a328d6cd02b3ae4305173457077876d99e14f32/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/internal/JavaViolationDecorator.java#L41-L42

https://docs.pmd-code.org/latest/pmd_languages_java.html#violation-decorators

Do we miss something here?

**Comment by oowekyala** [comment]
> Do we miss something here? Like something like "FULLY_QUALIFIED_CLASS_NAME"?

Yes. Maybe we can specify `CLASS_NAME` to mean eg here `Form.Property` for nested classes though.
